src/Diagnostics.ModelsAndUtils/Models/ResponseExtensions/Guage.cs

```diff
@@ -0,0 +1,231 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+using System.Data;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+
+namespace Diagnostics.ModelsAndUtils.Models.ResponseExtensions
+{
+    
+
+    public class Guage
+    {
+        /// <summary>
+        /// Create a new Guage instance. Does not have a default constructor.
+        /// </summary>
+        public Guage(InsightStatus status, double percentFilled, string displayValue, string label, GuageSize size, string description = "")
+        {
+            try
+            {
+                this.Status = status;
+                this.PercentFilled = percentFilled;
+                this.DisplayValue = displayValue;
+                this.Label = label;
+                if (size == GuageSize.Inherit) throw new Exception("Guage elements must have a size. Allowed size's are GuageSize.Small, GuageSize.Medium and GuageSize.Large. GuageSize.Inherit is allowed to use only with the Response.AddGuages extension method.");
```

Comment:
Does this mean It will throw exception every time I create a Guage object with Inherit Size?

Reply:
Yes, the inherit size is not supposed to be used with Guage object, it is supposed to be passed as a parameter to the AddGuages Response Extension method. I tried to mention its purpose in the comment. Is there a better way for me to achieve this ? Have a different enum perhaps ?